.github/workflows/fuzz.yml

```diff
@@ -19,6 +19,7 @@ jobs:
         uses: actions/checkout@v2
         with:
           persist-credentials: false
+          ref: master
```

Comment:
why is this change part of the commit

Reply:
I added this because the previous commit crashed the fuzz CI. The error occurred during repo checkout:
```
  /usr/bin/git config --local --name-only --get-regexp core\.sshCommand
  /usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
  Error: fatal: No url found for submodule path 'aws-c-common' in .gitmodules
  Error: The process '/usr/bin/git' failed with exit code 128
```
The checkout here should not use aws-c-common.

Reply:
doesn't this run fuzzer on master branch only, so there is no point of running it during PR? Or does it disable fuzz during PR?

Reply:
maybe @danblitzhou can look at this since he setup CI for fuzz

Reply:
I don't think it will disable fuzz during PR. However, the `ref` will checkout the master branch only.
```
    # The branch, tag or SHA to checkout. When checking out the repository that
    # triggered a workflow, this defaults to the reference or SHA for that event.
    # Otherwise, uses the default branch.
    ref: ''
```

Reply:
so this is the worst combination. fuzzer runs when PR is submitted, but not on the PR, thus hiding potential errors later on

Reply:
@LinerSu still waiting for resolution on this. I can't accept the PR in this form

Reply:
I have asked @danblitzhou about this. He suggests removing forced auth token to be not persisted. The default is persistent. Hope this works.